extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php

```diff
@@ -0,0 +1,432 @@
+
+			$sDigests .=  $wikiname . "\n";
+			if ( $usehtmlemail ) {
+				$sDigestsHTML .= "<b>" . $wikiname . "</b><br/>\n";
+			}
+
+			if ( !empty( $aDigest['pages'] ) ) {
+				if ( $usehtmlemail ) {
+					$sDigestsHTML .= "<ul>\n";
+				}
+
+				foreach ( $aDigest['pages'] as $aPageData ) {
+					// watchlist tracking, rt#33913
+					$url = $aPageData['title']->getFullURL( 's=dgdiff' . ( $aPageData['revisionId'] ? "&diff=" . $aPageData['revisionId'] . "&oldid=prev" : "" ) );
+
+					// plain email
+					$sDigests .= $url . "\n";
+
+					// html email
+					if ( $usehtmlemail ) {
+						$pagename = $aPageData['title']->getArticleName();
+						$pagename = str_replace( '_', ' ', rawurldecode( $pagename ) );
+						$sDigestsHTML .= '<li><a href="' . $url . '">' . $pagename . "</a></li>\n";
+					}
+
+					$iPagesCount++;
+				}
+
+				if ( $usehtmlemail ) {
+					$sDigestsHTML .= "</ul>\n<br/>\n";
+				}
+			}
+
+			# blog comments
+			if ( !empty( $aDigest['blogs'] ) ) {
+				foreach ( $aDigest['blogs'] as $blogTitle => $blogComments ) {
+					# $countComments = ($blogComments['comments'] >= $blogComments['own_comments']) ? intval($blogComments['comments'] - $blogComments['own_comments']) : $blogComments['comments'];
+					$countComments = $blogComments['comments'];
+
+					$tracking_url = $blogComments['blogpage']->getFullURL( 's=dg' ); // watchlist tracking, rt#33913
+
+					$message = wfMsgReplaceArgs(
+						( $countComments != 0 ) ? $this->getLocalizedMsg( 'globalwatchlist-blog-page-title-comment', $oUserLanguage ) : "$1",
+						array (
+							0 => $tracking_url, // send the ugly tracking url to the plain emails
+							1 => $countComments
+						)
+					);
+					$sDigestsBlogs .= $message . "\n";
+
+					if ( $usehtmlemail ) {
+						// for html emails, remake some things
+						$clean_url = $blogComments['blogpage']->getFullURL();
+						$clean_url = str_replace( '_', ' ', rawurldecode( $clean_url ) );
+						$message = wfMsgReplaceArgs(
+							( $countComments != 0 ) ? $this->getLocalizedMsg( 'globalwatchlist-blog-page-title-comment', $oUserLanguage ) : "$1",
+							array (
+								0 => "<a href=\"{$tracking_url}\">" . $clean_url . "</a>", // but use the non-tracking one for html display
+								1 => $countComments
+							)
+						);
+						$sDigestsBlogsHTML .= $message . "<br/>\n";
+					}
+
+					$iBlogsCount++;
+				}
+				$sDigestsBlogs .= "\n";
+			}
+
+			$sDigests .= "\n";
+		}
+		if ( $isDigestLimited ) {
+			$sDigests .= $this->getLocalizedMsg( 'globalwatchlist-see-more', $oUserLanguage ) . "\n";
+		}
+		$aEmailArgs = array(
+			0 => ucfirst( $oUser->getName() ),
+			1 => ( $iPagesCount > 0 ) ? $sDigests : $this->getLocalizedMsg( 'globalwatchlist-no-page-found', $oUserLanguage ),
+			2 => ( $iBlogsCount > 0 ) ? $sDigestsBlogs : "",
+		);
+
+		$sMessage = $this->getLocalizedMsg( 'globalwatchlist-digest-email-body', $oUserLanguage ) . "\n";
+		if (empty($aEmailArgs[2])) $sMessage = $this->cutOutPart($sMessage, '$2', '$3');
```

Comment:
These legacy code is quite in disagreement with our conventions. Maybe a quick auto-format of this file will do.